### PR TITLE
Fixed issue #18904: Stored XSS with Expression Script in End page

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/helpers/SurveyRuntimeHelper.php [hunk 1]
```diff
@@ -1313,11 +1313,18 @@ private function moveSubmitIfNeeded()
...
+            $validator = new LSYii_Validators();
             $this->aSurveyInfo['aCompleted']['sPluginHTML']  = implode("\n", $blocks) . "\n";
             $this->aSurveyInfo['surveyls_url']               = passthruReplace($this->aSurveyInfo['surveyls_url'], $this->aSurveyInfo);
             $this->aSurveyInfo['surveyls_url']               = $this->processString($this->aSurveyInfo['surveyls_url'], 3, 1);
+            if ($validator->isXssUrl($this->aSurveyInfo['surveyls_url'])) {
+                $this->aSurveyInfo['surveyls_url'] = "";
+            }
             $this->aSurveyInfo['aCompleted']['sSurveylsUrl'] = $this->aSurveyInfo['surveyls_url'];
             $this->aSurveyInfo['surveyls_urldescription'] = $this->processString($this->aSurveyInfo['surveyls_urldescription'], 3, 1);
+            if ($validator->isXssUrl($this->aSurveyInfo['surveyls_urldescription'])) {
```
Comment (Shnoulle) [inline application/helpers/SurveyRuntimeHelper.php:1325]: Here : not needed , it's a string shown. 
It's like the end text.

**Review by Shnoulle — CHANGES_REQUESTED** [review]
fix only the url, not the description (This one must be XSS like other text)

File: application/helpers/SurveyRuntimeHelper.php (hunk 1 above)
Comment (gabrieljenik) [reply application/helpers/SurveyRuntimeHelper.php]: Done.

**Comment by gabrieljenik** [comment]
@olleharstedt Added you to review as Denis is on vacations.

**Review by Shnoulle — APPROVED** [review]

**Comment by Shnoulle** [comment]
> @olleharstedt Added you to review as Denis is on vacations.

OK for me :)

**Pull request merged**